runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/Annotations.kt

```diff
@@ -58,3 +58,16 @@ public annotation class ExperimentalApi
  */
 @DslMarker
 public annotation class SdkDsl
+
+/**
+ * Indicates that the annotated API is deprecated and scheduled for removal in an upcoming minor version.
+ * Any code using this API should migrate to the suggested replacement as soon as possible.
+ */
+@RequiresOptIn(
+    message = "This API is deprecated and will be removed in an upcoming minor version.",
```

Comment:
nit: Hopefully this doesn't happen for a while, but we may make a _major_ version bump that deletes APIs, so this message should probably just say "... scheduled for removal in an upcoming ~~minor~~ version"